01-node-tutorial/answers/11-fs-async.js

```diff
@@ -0,0 +1,32 @@
+const fs = require('fs');
+
+const textFileLocation = 'temporary\\fileB.txt';
+const test = 'I am an async that syncs async\'fully.';
+const text = 'If an async is sync\'ed sync\'fully';
+const testText = 'is ain\'t sync\'ing async\'fully if at all';
+let flag = '';
+
+for (let i = 0; i < 3; i++) {
+  if (i) flag = {flag: 'a'};
+  fs.writeFileSync(
+    textFileLocation, 
+    `${test} ${text} ${testText} #${i + 1}!\n`, 
+    flag,
+    (err, result) => {
+      if (err) {
+        console.log(err);
+        return;
+      };
+      console.log(result);
```

Comment:
In the lesson, the requirement is to put the code that writes the next line within the callback itself. Instead of using a for loop to call the function outside a callback each time, what would you do to ensure the next line is written from within the callback itself?

Reply:
Oh my, only now I noticed I used the sync method. Here is what the code was supposed to look like. Can you please check if this one is fine? If it is, do you want me to push it or I can push it with the next PR?

```
const fs = require('fs');

const textFileLocation = 'temporary\\fileB.txt';
const text = 'I am an async that syncs async\'fully. If an async is sync\'ed sync\'fully it ain\'t sync\'ing async\'fully if at all\n';
let flagA = { flag: 'a' };

fs.writeFile(textFileLocation, `1 ${text}`, (err, result) => {
  if (err) {
    console.log(err);
    return;
  };
  fs.writeFile(textFileLocation, `2 ${text}`, flagA, (err, result) => {
    if (err) {
      console.log(err);
      return;
    };
    fs.writeFile(textFileLocation, `3 ${text}`, flagA, (err, result) => {
      if (err) {
        console.log(err);
        return;
      };
      fs.readFile(textFileLocation, 'utf8', (err, result) => {
        if (err) {
          console.log(err);
          return;
        };
        console.log(`Reading file:\n${result}`);
      });
    });
  });
});
```

Reply:
You should reopen the PR and commit and push the fixes, but do not merge yet. That way I can look over the next iteration properly. DM me if you need anything

Reply:
Done, https://github.com/SergeDot/node-express-course/pull/3